recipes/glib/all/conanfile.py

```diff
@@ -48,6 +48,10 @@ def configure(self):
                 "glib can not be built as static library on Windows. "
                 "see https://gitlab.gnome.org/GNOME/glib/-/issues/692"
             )
+        if tools.Version(self.version) < "2.67.0":
+            if not self.options.with_elf:
+                self.output.warn("libelf dependency can't be disabled in glib < 2.67.0")
+            del self.options.with_elf
```

Comment:
```suggestion
        if tools.Version(self.version) < "2.67.0" and not self.options.with_elf:
            raise ConanInvalidConfiguration("libelf dependency can't be disabled in glib < 2.67.0")        
```

This cause should treated on `config_options` instead. As it's an option which is already in the official package, it will better blocking the package id instead of removing it.

Reply:
Like this?
```python
    def config_options(self):
        [...]
        if tools.Version(self.version) < "2.67.0":
            del self.options.with_elf
```

Reply:
Yes, but as this option is already inserted in the official recipe you have to consider it illegal due that minimal version, if you remove it, for who is using, Conan it will raise an ConanException, because it's no longer valid. Instead, you can raise an ConanInvalidConfiguration, which better in terms of filtering for CI, and it will not change the package ID, as the option is still there.